### PR TITLE
Refactor SingleSubst format determination

### PR DESCRIPTION
**Pull request by simoncozens**
I had to look up what a `for...else...` loop meant. It's really not obvious. This is hopefully a bit easier to follow.

### CODE REVIEW

**Comment by anthrotype** [comment]
> I had to look up what a for...else... loop meant

**Comment by anthrotype** [comment]
> I had to look up what a for...else... loop meant

I just read the `else` as if it's written `ifnobreak`

**Comment by justvanrossum** [comment]
The rationale of "let's refactor because it's using a Python feature I haven't seen before" is rather dubious.

It's an arbitrary change that perhaps makes it better for you, perhaps not for others. It's completely siubjective, and therefore better left alone.

File: Lib/fontTools/ttLib/tables/otTables.py [hunk 1]
```diff
@@ -752,20 +752,17 @@ def preWrite(self, font):
...
-			else:
-				format = 1
+			format = 1
+			items = iter(gidItems)
+			firstInID, firstOutID = next(items)
+			delta = (firstOutID - firstInID) % 65536
+			for inID, outID in items:
+				if (inID + delta) % 65536 != outID:
+					format = 2
+					break
```
Comment (anthrotype) [inline Lib/fontTools/ttLib/tables/otTables.py:765]:
```python
format = 2 if any((inID + delta) % 65536 != outID for inID, outID in items) else 1
```

**Review by anthrotype — APPROVED** [review]

**Comment by anthrotype** [comment]
> perhaps makes it better for you, perhaps not for others

fair point. I actually had to read this several times to convince myself it was doing the same thing as before the patch.

**Pull request closed without merging**